### PR TITLE
Added Docker support for server only mode

### PR DESCRIPTION
**Pull request by bastilimbach**
I added a Dockerfile and some additional docker related files to support the deployment of the server only mode using [Docker](https://docker.com).

@MichMich to make the shell command work you need to create an account at [Docker Hub](https://hub.docker.com) and link this repository to it. So every time the Github Repo gets updated, the Docker Image will be rebuild and updated automatically.

Is it correct to add my changes under the "## [2.1.0] - 2016-12-31" section of the changelog?

- [x] Add the change to CHANGELOG.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,41 @@ curl -sL https://raw.githubusercontent.com/MichMich/MagicMirror/master/installer
...
+```javascript
+var config = {
+	ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:172.17.0.1"]
+};
+```
+
+#### Manual
+
+1. Download and install the latest Node.js version.
+2. Clone the repository and check out the beta branch: `git clone https://github.com/MichMich/MagicMirror`
```
Comment (roramirez) [inline README.md]: Should set master branch here not beta.

Comment (bastilimbach) [inline README.md]: What do you mean? Where did I use the "beta" branch?

File: README.md [hunk 2]
```diff
@@ -46,8 +46,41 @@ curl -sL https://raw.githubusercontent.com/MichMich/MagicMirror/master/installer
 **Note:** if you want to debug on Raspberry Pi you can use `npm start dev` which will start the MagicMirror app with Dev Tools enabled.
 
 ### Server Only
+In some cases, you want to start the application without an actual app window. In this case, you can start MagicMirror² in server only mode. This will start the server, after which you can open the application in your browser of choice.
```
Comment (roramirez) [inline README.md]: No remove reference to command `node serveronly` some users can be not understand  how run the mode server only.

File: README.md (hunk 1 above)
Comment (roramirez) [inline README.md]: Yes, beta branch isn't use anymore. When you do it a checkout from url is used master branch. Just change the beta by master.

This is  change done  few days ago
https://github.com/MichMich/MagicMirror/commit/a591cf1d21813b8358245691078f1a49cfcc04e0

**Comment by MichMich** [comment]
@bastilimbach Please add your changes to the `[2.1.1] - Unreleased` section of the CHANGELOG. Aso, the change doesn't seem to pass the travis test. Please run grunt to check what's wrong.

Thanks for your contribution!

File: README.md (hunk 1 above)
Comment (bastilimbach) [inline README.md]: Oh now I see what you mean. Sorry. I just copied it from line 40. Should I change it there too?

Comment (roramirez) [inline README.md]: Yes.

File: js/defaults.js [hunk 3]
```diff
@@ -8,7 +8,7 @@
  */
 
 var defaults = {
-	port: 8080,
+	port: process.env.MM_PORT || 8080,
```
Comment (bastilimbach) [inline js/defaults.js]: @MichMich I did run grunt before and couldn't figure out what the problem was. jsLint says there are irregular whitespaces but there aren't any. Or am I completely wrong here?

Comment (roramirez) [inline js/defaults.js]: After MM_PORT there a stranger space. Irregular whitespace. In character 30. Add a normal space.

Comment (bastilimbach) [inline js/defaults.js]: Strange. But it worked. Thanks 👍

Comment (roramirez) [inline js/defaults.js]: _Various whitespace characters can be inputted by programmers by mistake for example from copying or keyboard shortcuts. Pressing Alt + Space on OS X adds in a non breaking space character for example._
http://eslint.org/docs/rules/no-irregular-whitespace

**Comment by MichMich** [comment]
Nice work! Thanks!

**Comment by morozgrafix** [comment]
@bastilimbach Looks like this change is causing black screen on load when starting MM with `npm start` or `node serveronly` and following error in the console log of chrome/electron.

```
Uncaught ReferenceError: process is not defined
    at defaults.js:11
(anonymous) @ defaults.js:11
```

I believe that culprit is https://github.com/MichMich/MagicMirror/commit/354b745c39a19915c3a8547445a5049555bd37ff#diff-75953debd6cf3faf33d4ff10e8353645R11 

More details https://forum.magicmirror.builders/post/13221

**Comment by fewi** [comment]
> @MichMich to make the shell command work you need to create an account at Docker Hub and link this repository to it. So every time the Github Repo gets updated, the Docker Image will be rebuild and updated automatically.

Any progress on this? Would be nice to get updated Docker images at Docker Hub

**Comment by bastilimbach** [comment]
@fewi Version 2.1.1 isn't released yet. Look at the [Changelog](https://github.com/MichMich/MagicMirror/blob/develop/CHANGELOG.md). I think with the release of 2.1.1, @MichMich will put the image onto Docker Hub.

**Comment by MichMich** [comment]
@fewi For now you can just use the `develop` branch.

**Comment by bastilimbach** [comment]
@fewi @MichMich Yeah, you can checkout the `develop` branch and build the image yourself by running:
```shell
docker build -t magic_mirror .
```
After building the image, run a container with this image by executing:
```shell
docker run  -d \
            --publish 80:8080 \
            --restart always \
            --volume ~/magic_mirror/config:/opt/magic_mirror/config \
            --volume ~/magic_mirror/modules:/opt/magic_mirror/modules \
            --name magic_mirror \
            magic_mirror
```
You might want to change the volume paths.

**Comment by fewi** [comment]
Yesterday I was trying to build & run it on my windows machine. But it did not work. I made a PR to fix this https://github.com/MichMich/MagicMirror/pull/743

**Comment by AnyKey8472** [comment]
Hello everybody, 
I tried to get this docker image to run on my 6.1.1 Synology DS, but strugled with the installation steps. After downloading it via synology gui and (what I first thought was the installation) unpacking it, I don't know how to run the above mentioned code. 
I'm new to docker in generell and a little bit overwhelmed, is there a readme, or tutorial, how to setup your docker container on a synology system? 
Thanks
Andre

**Pull request merged**